Azure/azure-cli
# {Core} Rename mock.py

**Description**  

We are working on running tests for installed `azure-cli` packages. When running pytest for tests in `azure-cli-core` in MSI installed `azure-cli`, it will insert `C:\\Program Files (x86)\\Microsoft SDKs\\Azure\\CLI2\\Lib\\site-packages\\azure\\cli\\core` in the beginning of `sys.path`. For `import mock`, it finds and returns `C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\core\mock.pyc` instead of `C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\lib\site-packages\mock`.

This PR renames `mock.py` to `mock_cli.py` to avoid the import conflicts.

**Testing Guide**  

Pull the changes:
`git pull https://github.com/fengzhou-msft/azure-cli.git fix_core_mock`

On Git Bash:
`cd azure-cli`
`. scripts/ci/build.sh`

This script builds `azure-cli` related wheels including the `azure_cli_testsdk-0.2.4.dev*-py3-none-any.whl` and `azure_cli_fulltest-1.0.0.dev*-py2.py3-none-any.whl` in `azure-cli\artifacts\build`. The latter one includes all the tests from `azure-cli` repository.

On a Powershell with Admin privilege:

`& 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe' -m pip install pytest`

`& 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe' -m pip install \path\to\azure-cli\artifacts\build\azure_cli_testsdk-0.2.4.dev*-py3-none-any.whl`

`& 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe' -m pip install  \path\to\azure-cli\artifacts\build\azure_cli_fulltest-1.0.0.dev*-py2.py3-none-any.whl`

`& 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe' -m pytest -x -v 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\core\tests\test_cloud.py::TestCloud::test_add_get_cloud_with_hybrid_profile' -s`

Without this PR, you'll see the error:

> AttributeError: module 'mock' has no attribute 'patch'

With this PR, the test would pass.



**History Notes**  


[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.
